diambra/arena/env_settings.py

```diff
@@ -0,0 +1,196 @@
+from dataclasses import dataclass
+from typing import Union, List, Tuple
+from diambra.arena.utils.gym_utils import available_games
+import numpy as np
+
+MAX_VAL = float("inf")
+MIN_VAL = float("-inf")
+
+def check_num_in_range(key, value, bounds):
+    assert (value >= bounds[0] and value <= bounds[1]), "\"{}\" ({}) value must be in the range {}".format(key, value, bounds)
+
+def check_val_in_list(key, value, valid_list):
+    assert (value in valid_list), "\"{}\" ({}) admissible values are {}".format(key, value, valid_list)
+
+
+@dataclass
+class EnvironmentSettings:
+
+    game_id: str
+
+    # System level
+    step_ratio: int = 6
+    disable_keyboard:bool = True
+    disable_joystick: bool = True
+    rank: int = 0
+    seed: int = -1
+    env_address: str = "localhost:50051"
+    grpc_timeout: int = 60
+
+    # Game level
+    player: str = "Random"
+    continue_game: float = 0.0
+    show_final: bool = True
+    difficulty: int = 3
+    frame_shape: Tuple[int, int, int] = (0, 0, 0)
+
+    tower: int = 3  # UMK3 Specific
+
+    # Environment level
+    hardcore: bool = False
+
+    def sanity_check(self):
```

Comment:
Consider making the members of enviromental setting specific types or use `Literal`(https://docs.python.org/3/library/typing.html#typing.Literal) to reduce the runtime checking you do here:

Reply:
Thanks for the suggestion, unfortunately Literal is available since python 3.8, and we still need 3.7 for stable baselines. I'll make a not in the code, we will drop 3.7 in the future and implement this.

Anyways it is just called once per execution, so no problem in terms of computational burden